cpp/ql/src/Security/CWE/CWE-129/ImproperArrayIndexValidation.ql

```diff
@@ -50,8 +46,12 @@ predicate hasUpperBoundsCheck(Variable var) {
 }
 
 predicate nodeIsBarrierEqualityCandidate(DataFlow::Node node, Operand access, Variable checkedVar) {
-  readsVariable(node.asInstruction(), checkedVar) and
-  any(IRGuardCondition guard).ensuresEq(access, _, _, node.asInstruction().getBlock(), true)
+  exists(VariableAccess va |
+    va = node.asExpr() and
+    va.getTarget() = checkedVar and
+    access.getDef().getAst() = va and
+    any(IRGuardCondition guard).ensuresEq(access, _, _, access.getDef().getBlock(), true)
+  )
```

Comment:
Is this actually equivalent? It looks like `access` and `node` were only tied together by the `Variable` previously, but now they're tied by the `VariableAccess`.